src/argilla/client/feedback/training/base.py

```diff
@@ -0,0 +1,288 @@
+                seed=self._seed,
+                model=self._model,
+            )
+        elif framework is Framework.SPACY:
+            from argilla.client.feedback.training.frameworks.spacy import (
+                ArgillaSpaCyTrainer,
+            )
+
+            self._trainer = ArgillaSpaCyTrainer(
+                feedback_dataset=self._dataset,
+                training_task_mapping=self._training_task_mapping,
+                prepared_data=self._prepared_data,
+                seed=self._seed,
+                model=self._model,
+                gpu_id=gpu_id,
+            )
+        elif framework is Framework.OPENAI:
+            from argilla.client.feedback.training.frameworks.openai import (
+                ArgillaOpenAITrainer,
+            )
+
+            self._trainer = ArgillaOpenAITrainer(
+                feedback_dataset=self._dataset,
+                training_task_mapping=self._training_task_mapping,
+                prepared_data=self._prepared_data,
+                seed=self._seed,
+                model=self._model,
+            )
+        elif framework is Framework.SPAN_MARKER:
+            if not isinstance(training_task_mapping, TrainingTaskMapingForTextClassification):
+                raise NotImplementedError(f"{Framework.SPAN_MARKER} only supports `TextClassification` tasks.")
+
+            from argilla.client.feedback.training.frameworks.span_marker import (
+                ArgillaSpanMarkerTrainer,
+            )
+
+            self._trainer = ArgillaSpanMarkerTrainer(
+                feedback_dataset=self._dataset,
+                training_task_mapping=self._training_task_mapping,
+                prepared_data=self._prepared_data,
+                seed=self._seed,
+                model=self._model,
+            )
+        else:
+            raise NotImplementedError(f"{framework} is not a valid framework.")
+
+        self._logger.info(self)
+
+    def __repr__(self) -> str:
+        """
+        `trainer.__repr__()` prints out the trainer's parameters and a summary of how to use the trainer
+
+        Returns:
+          The trainer object.
+        """
+        return f"""\
+ArgillaBaseTrainer info:
+_________________________________________________________________
+These baseline params are fixed:
+    dataset: {self._dataset}
+    task: {self._training_task_mapping}
+    train_size: {self._train_size}
+    seed: {self._seed}
+
+
+{self._trainer.__class__} info:
+_________________________________________________________________
+The parameters are configurable via `trainer.update_config()`:
+    {self._trainer}
+
+Using the trainer:
+_________________________________________________________________
+`trainer.train(output_dir)` to train to start training. `output_dir` is the directory to save the model automatically.
+`trainer.predict(text, as_argilla_records=True)` to make predictions.
+`trainer.save(output_dir)` to save the model manually."""
```

Comment:
In these blocks we could use `textwrap` which is a Python built-in tool to avoid that weird formatting from happening, while having the same output, without the `\t` 😄 

```suggestion
        return textwrap.dedent(
            f"""\
            ArgillaBaseTrainer info:
            _________________________________________________________________
            These baseline params are fixed:
                dataset: {self._dataset}
                task: {self._training_task_mapping}
                train_size: {self._train_size}
                seed: {self._seed}
            
            
            {self._trainer.__class__} info:
            _________________________________________________________________
            The parameters are configurable via `trainer.update_config()`:
                {self._trainer}
            
            Using the trainer:
            _________________________________________________________________
            `trainer.train(output_dir)` to train to start training. `output_dir` is the directory to save the model automatically.
            `trainer.predict(text, as_argilla_records=True)` to make predictions.
            `trainer.save(output_dir)` to save the model manually.
            """
        )
```

Reply:
Thanks! Did not know this one.